services/ark-api/ark-api/src/ark_api/api/v1/sessions.py

```diff
@@ -68,4 +69,139 @@ async def list_sessions(
         return SessionListResponse(
             items=all_sessions,
             total=len(all_sessions)
-        )
\ No newline at end of file
+        )
+
+
+@router.delete("/{session_id}")
+@handle_k8s_errors(operation="delete", resource_type="session")
+async def delete_session(
+    session_id: str,
+    namespace: Optional[str] = Query(None, description="Namespace for this request (defaults to current context)")
+) -> dict:
+    """Delete a specific session and all its messages."""
+    async with with_ark_client(namespace, VERSION) as client:
+        memory_dicts = await get_all_memory_resources(client)
+        
+        deleted_count = 0
+        
+        for memory_dict in memory_dicts:
+            memory_name = memory_dict.get("metadata", {}).get("name", "")
+            
+            try:
+                service_url = get_memory_service_address(memory_dict)
+                
+                # Make DELETE request to memory service
+                async with httpx.AsyncClient() as http_client:
+                    response = await http_client.delete(
+                        f"{service_url}/sessions/{session_id}",
+                        timeout=30.0
+                    )
+                    
+                    if response.status_code == 200:
```

Comment:
404 responses are silently treated as success. Consider propagating 404 to indicate session not found vs. silently succeeding. With RESTful design, 404 should clearly mean "session doesn't exist" not "route missing". Alternative: return success with metadata (deleted_count=0) so clients can inspect the result.

Suggestion:
```python
if response.status_code == 200:
    deleted_count += 1
elif response.status_code == 404:
    logger.debug(f"Session {session_id} not found in memory {memory_name}")
elif response.status_code == 500:
    raise HTTPException(
        status_code=500,
        detail=f"Failed to delete session {session_id} from database"
    )
```